Modify rule S2761: LaYC multiple unary operators (#2180)
SonarSource/rspec: rules/S2761/python/rule.adoc

== Why is this an issue?

> Calling the ``++not++`` or ``++~++`` prefix operator twice is usually a typo. The second invocation invalidates the first one:

Reviewer: ```suggestion
The repetition of a prefix operator (``++not++`` or ``++~++```) is usually a typo. The second operator invalidates the first one:
```

> Calling the ``++not++`` or ``++~++`` prefix operator twice might be redundant: the second invocation undoes the first. Such mistakes are typically typos. Either this is a bug if the operator was meant to be called once, or misleading if done on purpose.

Reviewer: ```suggestion
Calling the ``++not++`` or ``++~++`` prefix operator twice is usually a typo. The second invocation invalidates the first one:
```

[source,python]
----
a = False
b = ~~a # Noncompliant: same as a
----

> Calling ``++not++`` twice is sometimes done instead of using the dedicated `bool()` built-in function. However, the latter increases the code readability and should be preferred.

Reviewer: ```suggestion
While calling ``++not++`` twice is equivalent to calling the `bool()` built-in function, the latter increases the code readability, so it should be preferred.
```

[source,python]
----
a = 0

b = not not a # Noncompliant: use bool()
----

=== Exceptions

> If the ``++~++`` function has been overloaded in a customized class and has been called twice, no warning is raised as it is assumed to be the expected usage.

Reviewer: ```suggestion
The rule does not raise an issue if the ``++~++`` function is overloaded in a customized class, as it is assumed to be the expected usage.
```
